### ISSUE TITLE
Supervised Loss

### ISSUE THREAD

**Issue by gmbss0** [jsyoon0823/TimeGAN#25]
First of all, great work with TimeGAN! Really an interesting paper.

I have a question regarding the supervised loss, computed as follows:
`G_loss_S = tf.losses.mean_squared_error(H[:,1:,:], H_hat_supervise[:,:-1,:])` . 

I understand the purpose of the supervisor. However, I am confused by the fact that the generator is supposed to be trained on G_loss_S in the second step of training TimeGAN (you call it 'Training only with supervised loss'), although it does not contribute to the computation and consequently no gradients are available. Why is that?

**Comment by jsyoon0823** [comment]
Thank you for your interest in our paper.

As you know, we generate new samples using both generator and supervisor.
Generator generates E and Supervisor converts E to H.
So, we can think that the generator function is a combination of generator
and supervisor.

Therefore, you can think that the second stage of training is for
supervisor training which is a component of the overall generator.

Thanks!

**Comment by gmbss0** [comment]
Thanks for your response.

I understand...So, finally, you can confirm that the weigths of the generator (the generator computing E and not the combination of generator and supervisor) are **not** updated during the second stage of training?
Thanks!

**Comment by jsyoon0823** [comment]
Technically yes. I can confirm.
